addons/sourcemod/scripting/nd_building_fixes.sp

```diff
@@ -4,15 +4,25 @@
 #include <nd_rounds>
 #include <nd_struct_eng>
 #include <nd_stocks>
+#include <nd_commander_build>
 
 //Version is auto-filled by the travis builder
-public Plugin myinfo = 
+public Plugin myinfo =
 {
-	name 		= "[ND] Buildings Fixes",
-	author 		= "stickz",
-    description	= "Prevent building walls inside relays/repeaters",
-    version 	= "dummy",
-	url 		= "https://github.com/stickz/Redstone/"
+    name        = "[ND] Buildings Fixes",
+    author      = "stickz, databomb",
+    description = "Prevent building walls inside relays/repeaters",
+    version     = "dummy",
+    url         = "https://github.com/stickz/Redstone/"
+}
+
+// Check for dependency on nd_structure_intercept
+public void OnAllPluginsLoaded()
+{
+    if (!LibraryExists("nd_structure_intercept"))
+    {
+        SetFailState("Failed to find plugin dependency nd_structure_intercept");
```

Comment:
We don't need `SetFailState`. We are hooking a forward, which does not require a dependency. We should check if the library exists instead and use `ND_OnStructureCreated` instead if it doesn't.

Reply:
There should never be a case where we have to fallback on `ND_OnStructureCreated` as this would mean the game's server_srv.so file was updated in a way that broke the signature that's being detoured. This was originally added as a reminder to ensure that the nd_structure_intercept plugin is loaded. If the preference is to reserve `SetFailState` for situations that would trigger direct errors rather than a silent failure then we can remove the call here.

Reply:
See https://github.com/stickz/Redstone/commit/398c3cdea39952800872bbc88c20cad3b1dbacc2